.github/workflows/ci.yaml

```diff
@@ -241,13 +287,17 @@ jobs:
 
   publish-image:
     name: Publish built image
-    needs: [deploy-check, lint, golangci, unit-test, build-image-and-ensure-clean-branch]
+    needs: [definevars, deploy-check, lint, golangci, unit-test, build-image-and-ensure-clean-branch]
+    env:
+      IMAGE_TAG_BASE: ${{needs.definevars.outputs.imagetagbase}}
+      IMAGE_REPOSITORY: ${{needs.definevars.outputs.imagerepository}}
+      IMAGE_NAME: ${{needs.definevars.outputs.imagename}}
+      OPERATOR_SUGGESTED_NAMESPACE: ${{needs.definevars.outputs.operatorsuggestednamespace}}
     if: >
       (github.event_name == 'push') &&
       (github.ref == 'refs/heads/main' ||
        startsWith(github.ref, 'refs/heads/release-') ||
-       startsWith(github.ref, 'refs/tags/v')) &&
-      (github.repository == 'ramendr/ramen')
+       startsWith(github.ref, 'refs/tags/v'))
```

Comment:
Dropped repository requirements for publish, any fork can publish to their image repository, as long as their repository has the required github [secrets](https://docs.github.com/en/rest/actions/secrets?apiVersion=2022-11-28) defined.

Reply:
This breaks build in all forks:
https://github.com/nirs/ramen/actions/runs/4245524548/jobs/7381280367

We need to revert this change until we find a better solution.

Reply:
Ah! you pushed to main, understood. The job is failing as it is missing QUAY_ROBOT_TOKEN and QUAY_USERNAME secret.

There are 3 ways to go, add back the constraint and define a variable for the repository like others and use that, or ignore the fork action errors as it is a publish step and was not run till now, or add individual secrets to the forks.

IOW, this is not breaking the build, only the push to main and further the publish job.

Reply:
People are expected to keep their forks updated, so each time you sync to main
the test will run and fail to push.

I don't see a need to push images from forks, but having the option to do this
is nice. The issue is the default - if push was running only if the rpeo has
the required vars, this would work for any repo that want to push images, and 
will skip for other repos.

Reply:
@ShyamsundarR Do we have an issue to restore the repo name check? will it break downstream?
If not lets restore it and then find the best way to enable push on forks.

Reply:
I could make it work with secrects, since `job.<job_id>.if` cannot access secrets or environemnt variables. See why here:
https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability

But it can access vars, so I added a PUBLISH_IMAGES variable:
https://github.com/nirs/ramen/commit/612a8e7038129c0bb311b96f527f0b68e3478928

Example run when PUBLISH_IMAGES is not defined:
https://github.com/nirs/ramen/actions/runs/4246751910/attempts/4

Example run when PUBLISH_IMAGES is defined to 'true'
(expected to fail on my fork since no secrects, should work in this repo):
https://github.com/nirs/ramen/actions/runs/4246751910/jobs/7384060571

Reply:
See #723